### PR TITLE
Add compose files for grafana statistics

### PR DESCRIPTION
**Pull request by arsulegai**
There are two commits in this PR, one with newly introduced compose files for grafana. Another is providing option to limit number of transactions in a batch while sending workload.
1. Introduce a new compose file similar to adhoc/node.yaml under
tests, which enables telegraf statistics to be sent to influxdb.
2. Introduce a compose file to start grafana and influxdb services
which can be used to send statistics from sawtooth-raft adhoc
tests.

There's planned activity to update document, so not included with this PR.

### CODE REVIEW

**Comment by vaporos** [comment]
@arsulegai how can I test this?

**Comment by arsulegai** [comment]
> @arsulegai how can I test this?

**[Note]**
I added another commit to the same PR, to enable build and test behind proxy network.

We liked the way sawtooth-raft allows us to bring up variable number of validator nodes network. So this follows similar approach.

1. Bring up admin service
$ cd sawtooth-raft/
$ docker-compose -f admin.yaml up -d

2. Bring up grafana and influxdb services
$ cd sawtooth-raft/tests
$ SAWTOOTH_CORE=/path/to/repository/sawtooth-core docker-compose -f grafana.yaml up -d

3.  Bring up nodes as stated in sawtooth-raft documentation,
a. For example if it's non-genesis node
$ SAWTOOTH_CORE=/path/to/repository/sawtooth-core docker-compose -f  node-with-stats.yaml -p nongenesis up
b. For example if it's genesis node
$ SAWTOOTH_CORE=/path/to/repository/sawtooth-core GENESIS=1 docker-compose -f node-with-stats.yaml -p genesis up

4. To access grafana, go to browser and the link http://[IP-of-the-machine]:3000.

We do not need additional configuration changes for default files available in sawtooth-xxxx repositories.

**Comment by arsulegai** [comment]
> This is great--so this works with a network proxy and supports Grafana.
> From looking at the changes, the trick with Grafana is to run Telegraf in the Sawtooth Validator container.

Yes, I raised a PR in sawtooth-core to default enable telegraf start tracking "sawtooth" processes.

**Comment by arsulegai** [comment]
@vaporos Could you please have a look into this PR?

File: adhoc/workload.yaml [hunk 1]
```diff
@@ -42,6 +42,7 @@ services:
         --key-file /shared_data/keys/workload.priv \
         --rate ${RATE:-1} \
         --urls $$TARGETS \
+        --batch-size ${BATCH:-1} \
```
Comment (peterschwarz) [inline adhoc/workload.yaml]: This should be in a separate commit, with a justification for the change in the commit message.

Comment (arsulegai) [reply adhoc/workload.yaml]: Ok

**Comment by arsulegai** [comment]
> The proxy support should probably be in a separate PR.

Ok

File: adhoc/workload.yaml (hunk 1 above)
Comment (danintel) [reply adhoc/workload.yaml]: I see this change is in a separate commit: _Allow limiting number of transactions in a batch for workload_

**Review by vaporos — APPROVED** [review]
